File: sourceCode/Windows7-Jumplist-cs/Backup/Windows7Jumplist/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.IO;
using Microsoft.WindowsAPICodePack.Taskbar;
using Microsoft.WindowsAPICodePack.Shell;
using System.Reflection;

namespace Windows7Jumplist
{
    public partial class MainForm : Form
    {
        // Keep a reference to the Taskbar instance
        private TaskbarManager windowsTaskbar = TaskbarManager.Instance;

        public MainForm()
        {
            InitializeComponent();
        }

        private void MainForm_Shown(object sender, EventArgs e)
        {
            if (TaskbarManager.IsPlatformSupported)
            {
                CreateJumpList();

                windowsTaskbar.SetProgressState(TaskbarProgressBarState.NoProgress);
            }

            Program.HandleCmdLineArgs();
        }

        
        protected override void WndProc(ref Message m)
        {
            if (m.Msg == WindowsMessageHelper.ARG1)
            {
                lblMessage.Text = "Arg1"; 
                lblMessage.BackColor = Color.LightGreen;
            }
            else if (m.Msg == WindowsMessageHelper.ARG2)
            {
                lblMessage.Text = "Arg1";
                lblMessage.BackColor = Color.LightBlue;
            }
            else if (m.Msg == WindowsMessageHelper.ARG3)
            {
                lblMessage.Text = "Arg1";
                lblMessage.BackColor = Color.Yellow;
            }
            else base.WndProc(ref m);
        }

        private void CreateJumpList()
        {
            JumpList jl = JumpList.CreateJumpList();

            // Show user files: Recent, Frequent, or None
            jl.KnownCategoryToDisplay = JumpListKnownCategoryType.Recent;

            // Add my own links (nouns)
            JumpListCustomCategory catActions = new JumpListCustomCategory("Destinations");
            catActions.AddJumpListItems(
                new JumpListLink(Environment.GetFolderPath(Environment.SpecialFolder.MyPictures), "My Pictures"),
                new JumpListLink("http://blogs.msdn.com/coding4fun", "Visit Coding4Fun"),
                new JumpListLink("http://code.msdn.microsoft.com/WindowsAPICodePack", "Windows API Code Pack")
                //new JumpListItem(@"c:\Test1.c4f")
                );

            jl.AddCustomCategories( catActions);

            // Path to Windows system folder
            string systemFolder = Environment.GetFolderPath(Environment.SpecialFolder.System);

            // Add our user tasks (verbs)

            jl.AddUserTasks(new JumpListLink(Path.Combine(systemFolder, "notepad.exe"), "Open Notepad")
            {
                IconReference = new IconReference(Path.Combine(systemFolder, "notepad.exe"), 0)
            });

            jl.AddUserTasks(new JumpListLink(Path.Combine(systemFolder, "mspaint.exe"), "Open Paint")
            {
                IconReference = new IconReference(Path.Combine(systemFolder, "mspaint.exe"), 0)
            });

            jl.AddUserTasks(new JumpListSeparator());

            jl.AddUserTasks(new JumpListLink(Assembly.GetEntryAssembly().Location, "Action 1 (Green)")
            {
                IconReference = new IconReference(Assembly.GetEntryAssembly().Location, 0),
                Arguments = "-1"
            });

            jl.AddUserTasks(new JumpListLink(Assembly.GetEntryAssembly().Location, "Action 2 (Blue)")
            {
                IconReference = new IconReference(Assembly.GetEntryAssembly().Location, 0),
                Arguments = "-2"
            });

            jl.AddUserTasks(new JumpListLink(Assembly.GetEntryAssembly().Location, "Action 3 (Yellow)")
            {
                IconReference = new IconReference(Assembly.GetEntryAssembly().Location, 0),
                Arguments = "-3"
            });

            jl.Refresh();
        }

    }
}

File: sourceCode/Windows7-Jumplist-vb/Windows7Jumplist/MainForm.vb
Imports Microsoft.WindowsAPICodePack.Taskbar
Imports Microsoft.WindowsAPICodePack.Shell
Imports System.IO
Imports System.Reflection

Partial Public Class MainForm

    Inherits Form
    ' Keep a reference to the Taskbar instance
    Private windowsTaskbar As TaskbarManager = TaskbarManager.Instance

    Public Sub New()
        InitializeComponent()
    End Sub

    Private Sub MainForm_Shown(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Shown
        If TaskbarManager.IsPlatformSupported Then
            CreateJumpList()

            windowsTaskbar.SetProgressState(TaskbarProgressBarState.NoProgress)
        End If

        Program.HandleCmdLineArgs()
    End Sub


    Protected Overrides Sub WndProc(ByRef m As System.Windows.Forms.Message)
        If m.Msg = WindowsMessageHelper.ARG1 Then
            lblMessage.Text = "Arg1"
            lblMessage.BackColor = Color.LightGreen
        ElseIf m.Msg = WindowsMessageHelper.ARG2 Then
            lblMessage.Text = "Arg1"
            lblMessage.BackColor = Color.LightBlue
        ElseIf m.Msg = WindowsMessageHelper.ARG3 Then
            lblMessage.Text = "Arg1"
            lblMessage.BackColor = Color.Yellow
        Else
            MyBase.WndProc(m)
        End If
    End Sub

    Private Sub CreateJumpList()
        Dim jl As JumpList = JumpList.CreateJumpList()

        ' Show user files: Recent, Frequent, or None
        jl.KnownCategoryToDisplay = JumpListKnownCategoryType.Recent

        ' Add my own links (nouns)
        Dim catActions As New JumpListCustomCategory("Destinations")

        catActions.AddJumpListItems( _
            New JumpListLink(Environment.GetFolderPath(Environment.SpecialFolder.MyPictures), "My Pictures"), _
            New JumpListLink("http://blogs.msdn.com/coding4fun", "Visit Coding4Fun"), _
            New JumpListLink("http://code.msdn.microsoft.com/WindowsAPICodePack", "Windows API Code Pack"))
        'new JumpListItem(@"c:\Test1.c4f")

        jl.AddCustomCategories(catActions)

        ' Path to Windows system folder
        Dim systemFolder As String = Environment.GetFolderPath(Environment.SpecialFolder.System)

        ' Add our user tasks (verbs)
        jl.AddUserTasks(New JumpListLink(Path.Combine(systemFolder, "notepad.exe"), "Open Notepad") _
                        With {.IconReference = New IconReference(Path.Combine(systemFolder, "notepad.exe"), 0)})

        jl.AddUserTasks(New JumpListLink(Path.Combine(systemFolder, "mspaint.exe"), "Open Paint") _
                        With {.IconReference = New IconReference(Path.Combine(systemFolder, "mspaint.exe"), 0)})

        jl.AddUserTasks(New JumpListSeparator())

        jl.AddUserTasks(New JumpListLink(Assembly.GetEntryAssembly().Location, "Action 1 (Green)") _
                        With {.Arguments = "-1", .IconReference = New IconReference(Assembly.GetEntryAssembly().Location, 0)})

        jl.AddUserTasks(New JumpListLink(Assembly.GetEntryAssembly().Location, "Action 2 (Blue)") _
                        With {.Arguments = "-2", .IconReference = New IconReference(Assembly.GetEntryAssembly().Location, 0)})

        jl.AddUserTasks(New JumpListLink(Assembly.GetEntryAssembly().Location, "Action 3 (Yellow)") _
                        With {.Arguments = "-3", .IconReference = New IconReference(Assembly.GetEntryAssembly().Location, 0)})

        jl.Refresh()
    End Sub

End Class
